api/provider/aws/environment_scaler.go

```diff
@@ -1,14 +1,758 @@
 package aws
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"sort"
+	"strconv"
+
+	"github.com/zpatrick/go-bytesize"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/api/job"
+	"github.com/quintilesims/layer0/api/provider"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+var defaultPorts = []int{
```

Comment:
`defaultPorts` for what?

Reply:
@diemonster: these ports are always used on a new ec2 instance - but perhaps this warrants a comment. 
@sesh-kebab: I don't think you can have const `[]int`, however, I've seen people get around this by using a function:
```
func defaultPorts() []int {
    return ...
}
```

Reply:
What's the benefit of `func` over `var` here?

Reply:
@tlake two things: we don't want the ports in question to be mutable, so we should use a `const`. But we can't do `const []int`, so @zpatrick was suggesting a function that returns `[]int`.

Reply:
Aaaaaah, I understand. :+1: ✅